tests/foreman/ui/test_discoveryrule.py

```diff
@@ -218,8 +220,8 @@ def test_positive_list_host_based_on_rule_search_query(
         assert values['table'][0]['IP Address'] == ip_address
         assert values['table'][0]['CPUs'] == str(cpu_count)
         # auto provision the discovered host
-        session.discoveredhosts.apply_action('Auto Provision', [discovered_host['name']])
-        assert not session.discoveredhosts.search('name = "{}"'.format(discovered_host['name']))
+        result = target_sat.api.DiscoveredHost(id=discovered_host['id']).auto_provision()
+        assert f'provisioned with rule {discovery_rule.name}' in result['message']
```

Comment:
why changing this to API in UI module?

Reply:
No, we shouldn't be doing it, because we have real hosts as well on which  I want that warning to be caught from airgun if there is a flaw. Also, we are testing the UI auto_provision in other places as well , so its just a simple change.

Reply:
@adarshdubey-star Can you please share any test where we are testing the UI interface for auto_provision

Reply:
https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_discoveredhost.py#L152